app/controllers/blogs_controller.rb

```diff
@@ -3,13 +3,16 @@
 class BlogsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
 
-  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :set_blog, only: %i[edit update destroy]
 
   def index
     @blogs = Blog.search(params[:term]).published.default_order
   end
 
-  def show; end
+  def show
+    @blog = Blog.find(params[:id])
+    Blog.published.find(params[:id]) unless @blog.owned_by?(current_user)
```

Comment:
```suggestion
    @blog = Blog.hogehoge(current_user).find(params[:id])
```

みたいにscopeを使って一発で取ってこれるようになるとよりベターですね。
ちょっとSQLの知識が必要になりますが、チャレンジできそうですか？

Reply:
scopeを使うと色々な問い合わせ方ができるんですね 😳📝 
こうかな？という形に修正してみました。 9acc8a8